### PR TITLE
Feature/style css selectors

### PR DESCRIPTION
**Pull request by Aiga115**
Added id selectors on modal windows 'Create workspace' buttons and repository input fields when adding a new repository.

### CODE REVIEW

File: applications/osb-portal/src/components/header/Banner.tsx [hunk 1]
```diff
@@ -58,9 +58,9 @@ export const Banner = (props: any) => {
               <Button variant="outlined">Take the tour</Button>
             }
             {user === null ? (
-              <Button variant="outlined" onClick={handleSignup}>Sign up</Button>
+              <Button id='signup' variant="outlined" onClick={handleSignup}>Sign up</Button>
```
Comment (filippomc) [inline applications/osb-portal/src/components/header/Banner.tsx:61]: The naming should be consistent and indicate more about the object and functionality.

`signup-button` here would be appropriate also for consistency with other -button ids

File: applications/osb-portal/src/components/common/OSBSpliButton.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const OSBSplitButton = (props: OSBSplitButtonProps) => {
 
 
     return (
-        <Grid container={true} direction="column" alignItems="center">
+        <Grid id='osb-split-button' container={true} direction="column" alignItems="center">
```
Comment (filippomc) [inline applications/osb-portal/src/components/common/OSBSpliButton.tsx:46]: A few things to fix here: 
- a potentially reusable component should not define ids, only classes: ids must be guaranteed to be unique
- the name of the component is not really indicative (sorry I missed this in previous pr). Should be more something like OSBApplicationChooserButton

File: applications/osb-portal/src/components/header/Banner.tsx [hunk 3]
```diff
@@ -58,9 +58,9 @@ export const Banner = (props: any) => {
...
             ) : null}
-            <Button variant="outlined" onClick={() => history.push('/repositories')}>View repositories</Button>
+            <Button id='view-repositories-button' variant="outlined" onClick={() => history.push('/repositories')}>View repositories</Button>
             <Button variant="outlined" onClick={() => window.open("https://docs.opensourcebrain.org/OSBv2/Overview.html")}>More information</Button>
```
Comment (filippomc) [inline applications/osb-portal/src/components/header/Banner.tsx:64]: Why this button was not added an id or class?

File: applications/osb-portal/src/components/header/Header.tsx [hunk 4]
```diff
@@ -101,15 +101,16 @@ export const Header = (props: any) => {
           <Paper>
             <ClickAwayListener onClickAway={handleMenuClose}>
               <MenuList autoFocusItem={menuOpen} id="user-menu">
-                {<MenuItem onClick={handleMyAccount}>My account</MenuItem>}
+                {<MenuItem id='my-account-item' onClick={handleMyAccount}>My account</MenuItem>}
                 {<MenuItem onClick={handleAccountHelp}>Account help</MenuItem>}
                 {/* <MenuItem>Settings</MenuItem> */}
-                <MenuItem onClick={handleUserLogout}>Logout</MenuItem>
+                <MenuItem id='logout' onClick={handleUserLogout}>Logout</MenuItem>
```
Comment (filippomc) [inline applications/osb-portal/src/components/header/Header.tsx:107]: logout-menu-item

File: applications/osb-portal/src/components/header/Header.tsx [hunk 5]
```diff
@@ -101,15 +101,16 @@ export const Header = (props: any) => {
           <Paper>
             <ClickAwayListener onClickAway={handleMenuClose}>
               <MenuList autoFocusItem={menuOpen} id="user-menu">
-                {<MenuItem onClick={handleMyAccount}>My account</MenuItem>}
+                {<MenuItem id='my-account-item' onClick={handleMyAccount}>My account</MenuItem>}
```
Comment (filippomc) [inline applications/osb-portal/src/components/header/Header.tsx:104]: my-account-menu-item

File: applications/osb-portal/src/components/header/Header.tsx [hunk 6]
```diff
@@ -101,15 +101,16 @@ export const Header = (props: any) => {
...
                 {<MenuItem onClick={handleAccountHelp}>Account help</MenuItem>}
                 {/* <MenuItem>Settings</MenuItem> */}
-                <MenuItem onClick={handleUserLogout}>Logout</MenuItem>
+                <MenuItem id='logout' onClick={handleUserLogout}>Logout</MenuItem>
               </MenuList>
             </ClickAwayListener>
           </Paper>
         </Popper>
         <Button
+          id='user-name-button'
```
Comment (filippomc) [inline applications/osb-portal/src/components/header/Header.tsx:113]: user-menu-open-button

File: applications/osb-portal/src/components/repository/EditRepoDialog.tsx [hunk 7]
```diff
@@ -314,6 +314,7 @@ export const EditRepoDialog = ({
 
   return (
     <Dialog
+      id='repository-dialog-window'
```
Comment (filippomc) [inline applications/osb-portal/src/components/repository/EditRepoDialog.tsx:317]: repository-edit-dialog

File: applications/osb-portal/src/components/repository/EditRepoDialog.tsx [hunk 8]
```diff
@@ -332,6 +333,7 @@ export const EditRepoDialog = ({
           <Box className="wrap">
             <FormControl variant="outlined">
               <Select
+                id='repository-source-input-element'
```
Comment (filippomc) [inline applications/osb-portal/src/components/repository/EditRepoDialog.tsx:336]: nice!

File: applications/osb-portal/src/components/workspace/NewWorkspaceAskUser.tsx [hunk 9]
```diff
@@ -33,6 +33,7 @@ export default (props: any) => {
               </Button>
           </Box>
           <Button
+            id='signup'
```
Comment (filippomc) [inline applications/osb-portal/src/components/workspace/NewWorkspaceAskUser.tsx:36]: Signup was already used elsewhere. Beware id must be unique in a page!

**Review by filippomc — CHANGES_REQUESTED** [review]
Thanks @Aiga115 we are heading into the right direction but better do another round on this -- please follow the guidelines in this document https://docs.google.com/document/d/1xZ00dl5tU-A01Beik4ed-inDVMVz_cAbQvP4coeDsRk/edit.

Basically must use very indicative names especially when it comes to ids, which should be guaranteed to be unique.

Practically speaking, my general suggestion is to specify ids only at layout level, never on potentially reusable components.
Classes can be used when in doubt, still with the same attention on naming.
Also, naming should be consistent across the application

File: applications/osb-portal/src/components/header/Banner.tsx (hunk 3 above)
Comment (Aiga115) [reply applications/osb-portal/src/components/header/Banner.tsx:64]: Do you mean 'more information' button?

File: applications/osb-portal/src/components/header/Header.tsx [hunk 10]
```diff
@@ -101,10 +101,10 @@ export const Header = (props: any) => {
           <Paper>
             <ClickAwayListener onClickAway={handleMenuClose}>
               <MenuList autoFocusItem={menuOpen} id="user-menu">
-                {<MenuItem onClick={handleMyAccount}>My account</MenuItem>}
+                {<MenuItem id='my-account-menu-item' onClick={handleMyAccount}>My account</MenuItem>}
```
[the remark this answers is not in the document]
Comment (filippomc) [reply applications/osb-portal/src/components/header/Header.tsx:104]: @Aiga115 still menu items having id

File: applications/osb-portal/src/components/workspace/NewWorkspaceToolBox.tsx [hunk 11]
```diff
@@ -67,8 +67,9 @@ export const WorkspaceToolBox = (props: any) => {
           </Grid>
           <Grid item={true} xs={12} sm={12} md={8} lg={9}>
             <Grid container={true} direction="row" justify="space-between" spacing={5} >
-              <Grid item={true} xs={12} sm={3} className={classes.justifyCenter} >
+              <Grid id='computational-modeling-item' item={true} xs={12} sm={3} className={classes.justifyCenter} >
```
Comment (filippomc) [inline applications/osb-portal/src/components/workspace/NewWorkspaceToolBox.tsx:70]: here a class would be preferable as the component is potentially reusable. But it's not, so can be fine

File: applications/osb-portal/src/components/workspace/Workspaces.tsx [hunk 12]
```diff
@@ -176,13 +176,13 @@ export const Workspaces = ({ user, counter }: any) => {
               />
               : null
             }
-            <Tab
+            <Tab id="featured-tab"
```
Comment (filippomc) [inline applications/osb-portal/src/components/workspace/Workspaces.tsx:179]: Another potentially reusable component with ids inside

File: applications/osb-portal/src/pages/HomePage.tsx [hunk 13]
```diff
@@ -33,27 +33,35 @@ export default (props: any) => {
         <Grid item={true} xs={12} sm={12} md={6} direction="column" className="verticalFill">
           <Box display="flex" >
             <Paper className={classes.paper} elevation={0}>
-              <Banner />
+              <div id='homepage-banner'>
```
Comment (filippomc) [inline applications/osb-portal/src/pages/HomePage.tsx:36]: Nice!

**Review by filippomc — CHANGES_REQUESTED** [review]
Dropped a few comments. The only change required is using classes instead of ids on menu items

File: applications/osb-portal/src/components/header/Banner.tsx (hunk 3 above)
Comment (filippomc) [reply applications/osb-portal/src/components/header/Banner.tsx]: the button after that one, line 64

**Pull request merged**